src/hedera/hederaEngine.js

```diff
@@ -0,0 +1,347 @@
+/**
+ * Created by Austin Bonander <austin@launchbadge.com> on 9/30/19.
+ */
+// @flow
+
+import * as hedera from '@hashgraph/sdk'
+import { bns } from 'biggystring'
+// import { currencyInfo } from './hederaInfo.js'
+import {
+  type EdgeCurrencyEngineOptions,
+  type EdgeCurrencyTools,
+  type EdgeIo,
+  type EdgeSpendInfo,
+  type EdgeTransaction,
+  type EdgeWalletInfo,
+  NoAmountSpecifiedError
+} from 'edge-core-js/types'
+
+import { CurrencyEngine } from '../common/engine.js'
+import { HederaPlugin } from '../hedera/hederaPlugin.js'
+
+export class HederaEngine extends CurrencyEngine {
+  hederaPlugin: HederaPlugin
+  client: ?hedera.Client
+  privateKey: hedera.Ed25519PrivateKey
+  otherMethods: Object
+  io: EdgeIo
+
+  constructor(
+    currencyPlugin: HederaPlugin,
+    walletInfo: EdgeWalletInfo,
+    opts: EdgeCurrencyEngineOptions,
+    io: EdgeIo
+  ) {
+    super(currencyPlugin, walletInfo, opts)
+    this.log('hedera constructor called', walletInfo)
+    this.hederaPlugin = currencyPlugin
+
+    const { hederaPrivateKey } = walletInfo.keys
+
+    this.privateKey =
+      hederaPrivateKey instanceof hedera.Ed25519PrivateKey
+        ? hederaPrivateKey
+        : hedera.Ed25519PrivateKey.fromString(hederaPrivateKey)
+
+    this.io = io
+
+    this.otherMethods = {
+      getAccountActivationQuote: async (params: Object) => {
+        const { currencyCode, activePublicKey } = params
+        if (!currencyCode || !activePublicKey) {
+          throw new Error('ErrorInvalidParams')
+        }
+
+        const options = {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            currency: currencyCode,
+            public_key: activePublicKey
+          })
+        }
+
+        const response = await (
+          await io.fetch('https://api.kabuto.sh/v1/account', options)
+        ).json()
+
+        this.walletLocalData.otherData.activationRequestId = response.id
+        this.walletLocalDataDirty = true
+
+        return {
+          paymentAddress: response.payment_addr,
+          currencyCode,
+          amount: bns.toFixed(response.fee, 3, 9),
+          otherParams: {
+            data: response.memo,
+            requestId: response.id,
+            statusUrl: response.statusUrl
+          },
+          exchangeAmount: '0'
+        }
+      },
+      submitActivationPayment: async (
+        txn: EdgeTransaction,
+        otherParams: Object
+      ) => {
+        const { requestId } = otherParams
+        const options = {
+          method: 'POST',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            txn_id: txn.txid
+          })
+        }
+        const response = await this.io.fetch(
```

Comment:
We talked about it internally and we're going to revisit the implementation of the creator so it accepts signed transactions and validates then submits them itself so that part won't be the client's responsibility anymore.